File: 1944-number-of-visible-people-in-a-queue/1944-number-of-visible-people-in-a-queue.py
class Solution:
    def canSeePersonsCount(self, heights: List[int]) -> List[int]:
        ans = [0] * len(heights)
        
        stack = []
        
        for i in range(len(heights)):
            while stack and heights[stack[-1]] <= heights[i]:
                ans[stack.pop()] += 1
            
            if stack:
                ans[stack[-1]] += 1
            stack.append(i)
        
        return ans

File: 1944-number-of-visible-people-in-a-queue/1944-number-of-visible-people-in-a-queue.cpp
class Solution {
public:
    vector<int> canSeePersonsCount(vector<int>& heights) {
        stack<int> *st = new stack<int>;
        vector<int> ans(heights.size(), 0);
        
        for (int i = 0; i < heights.size(); ++i){
            while (!st -> empty() && heights[st -> top()] <= heights[i]){
                ++ans[st -> top()];
                st -> pop();
            }
            if (!st -> empty()) ++ans[st -> top()];
            
            st -> push(i);
        }
        return ans;
    }
};